### PR TITLE
Stepic API -- example

### PR DESCRIPTION
**Pull request by chisler**
Example I added is to show that making and processing API-request is simple and may be interesting.

Python code uploads some lessons and makes three plots: 
1. (X = lesson_duration, Y = quantity_of_people_who_viewed_this_lesson)
2. (X = lesson_duration, Y = quantity_of_people_who_passed_this_lesson)
3. (X = lesson_duration, Y = difference_between_viewed_and_passed)

As a result, 5 minutes work allows us to see some real data and evaluate risk-duration of lesson that
goes with a biggest quantity of people who quit. (We don't normalize quantity of people who quit because we care about quantity).

### CODE REVIEW

File: examples/lessons_api_example (copy).py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Run with Python 3
```
Comment (psviderski) [inline examples/lessons_api_example (copy).py]: Does this file make sense?

**Comment by mehanig** [comment]
It doesn't working on my machine, maybe specific versions of packages needed? (pylab)

`Traceback (most recent call last):
  File "lessons_api_example.py", line 35, in <module>
    pylab.subplot (3, 1, 1)
AttributeError: 'module' object has no attribute 'subplot'`

File: examples/lessons_api_example (copy).py (hunk 1 above)
Comment (chisler) [inline examples/lessons_api_example (copy).py]: No, it's an untracked copy.

**Comment by chisler** [comment]
I've checked this out on my machine and it worked because of anaconda installed. 

I've pushed anaconda-independent version on my fork.

File: examples/lessons_api_example.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+client_id = "put_yours_here"
+client_secret = "put_yours_here"
+
+# 2. Get a token
+auth = requests.auth.HTTPBasicAuth(client_id, client_secret)
+resp = requests.post('https://stepic.org/oauth2/token/',
+                     data={'grant_type': 'client_credentials'},
+                     auth=auth
+                     )
+token = json.loads(resp.text)['access_token']
```
Comment (mehanig) [inline examples/lessons_api_example.py]: requests has .json() method , can be used here instead of json.loads() , like here - ef7efe2

**Comment by mehanig** [comment]
Looks good!
But plot still looks ambiguous to me( 
Can you make it more clear?
Also, pls fix json part, before I merge it

**Comment by chisler** [comment]
I'll answer and fix it it in 3 hours.

**Comment by chisler** [comment]
Hi! I've pushed a bit ad hoc but totally working on your machine code. It draws the same plots in text.

Matplotlib gives something like this:
quit

I had the same problem as you with matplotlib but installing some dev-pack helped for me.

File: examples/lessons_api_example.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
+# Run with Python 3
+import requests
+import pandas as pd
+import math
+
+
+class figure():
```
Comment (mehanig) [inline examples/lessons_api_example.py]: pep8 refers to start all classnames with Capital, so this should be `class Figure()`

**Comment by mehanig** [comment]
Good! 
Merged.
Also changed `figure` class to `Figure`

**Comment by chisler** [comment]
Cool, thanks!

**Comment by taery** [comment]
@chisler, можете еще сделать простенькую html-страницу, отображающую, что-нибудь из результатов вашего pull-реквеста? Например, тот же график или просто список уроков.

**Comment by chisler** [comment]
@taery, конечно! Сейчас у меня экзамен. Добавлю через пару часов.

**Comment by taery** [comment]
@chisler, не обязательно прям уж так спешить, удачи на экзамене =)

**Comment by chisler** [comment]
@taery, спасибо большое! Все прошло удачно.

Обновил [реп](https://github.com/chisler/Stepic-API/blob/master/examples/lessons_api_example.py). Переделал класс-построитель, добавил вывод в html доступных уроков и графиков. Шаблонизаторы я не использовал, чтобы не портить общий репозиторий.

Построитель графиков увеличил файл -- я постарался описать так, чтобы это действительно можно было использовать в качестве примера.

**Comment by taery** [comment]
@chisler, давайте это тоже пул-реквестом =)

**Comment by chisler** [comment]
Сделал запрос!

**Pull request merged**